extensions/Dockerfile

```diff
@@ -12,7 +12,7 @@ RUN --mount=type=secret,id=yumrepos,target=/os/secret.repo extensions/build.sh
 ## Creates the repo metadata for the extensions.
 ## This uses Fedora as a lowest-common-denominator because it will work on
 ## current p8/s390x. See https://github.com/openshift/os/issues/1000
-FROM quay.io/fedora/fedora:41 as builder
+FROM quay.io/fedora/fedora:42 as builder
```

Comment:
@jlebon @travier - should we consider switching back to centos stream9 here?

Reply:
Should be OK if we only build RHEL9 now. I think that's the case but I've not 100% verified.